src/ripple/app/hook/Enum.h

```
@@ -44,9 +46,14 @@ maxHookParameterValueSize(void)
 }
 
 inline uint32_t
-maxHookStateDataSize(void)
+maxHookStateDataSize(uint16_t hookStateScale)
 {
-    return 256U;
+    if (hookStateScale == 0)
+    {
+        // should not happen, but just in case
+        return 256U;
+    }
+    return 256U * hookStateScale;
```

we could add an extra defensive check here with a constant, I guess

---

like this?
```cpp
inline uint32_t
maxHookStateDataSize(uint16_t hookStateScale)
{
    if (hookStateScale == 0)
    {
        // should not happen, but just in case
        return 256U;
    }
    if (hookStateScale > maxHookStateScale())
    {
        // should not happen, but just in case
        return 256 * maxHookStateScale();
    }
    return 256U * hookStateScale;
}
```

---

I think assert/throw "fail fast" because it's a violated invariant indicating something went awry ?

---

I think it's too much to add that kind of processing here.

---

Well, we can do "return 256U * hookStateScale;" if it's just a simple calculation, but if you want to handle strange cases (0 || > max_scale) this way, it means you're explicitly washing over bad state. If it should not happen, assert that it doesn't happen. This is some weird middle ground, no?